source/MaterialXCore/Document.cpp

```diff
@@ -111,9 +111,28 @@ class Document::Cache
                 if (!nodeDefString.empty())
                 {
                     InterfaceElementPtr interface = elem->asA<InterfaceElement>();
-                    if (interface && (interface->isA<Implementation>() || interface->isA<NodeGraph>()))
+                    if (interface)
                     {
-                        implementationMap.emplace(interface->getQualifiedName(nodeDefString), interface);
+                        if (interface->isA<NodeGraph>())
```

Comment:
Would it be possible remove the use of the "nodedef" attribute on nodegraphs altogether, so associations through the <implementation> element is the one and only way to specify functional nodegraph for a nodedef?

Document upgrade would have to handle this, but then we have a single consistent way of handling implementations everywhere else in the code base.

Reply:
This is meant to be incremental and not remove any previous support for now. For the most part the branching is very isolated and the user won't see the different.

The upgrade path, I haven't looked into this yet, and don't want to do this yet as we still don't have a way to determine how to identify a patch release (a 1.38.3) from purely data -- i.e. avoid issues ran into last time we did forward upgrades.